presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/CountQueryPageSource.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.elasticsearch;
+
+import io.prestosql.elasticsearch.client.ElasticsearchClient;
+import io.prestosql.spi.Page;
+import io.prestosql.spi.connector.ConnectorPageSource;
+import io.prestosql.spi.connector.ConnectorSession;
+
+import static io.prestosql.elasticsearch.ElasticsearchQueryBuilder.buildSearchQuery;
+import static java.util.Objects.requireNonNull;
+
+class CountQueryPageSource
+        implements ConnectorPageSource
+{
+    private static final int BATCH_SIZE = 10000;
```

Comment:
please capture some motivation how we ended up with a modest number as a comment